core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -0,0 +1,746 @@
+            DBIO.successful()
+
+        } yield job.map(_.copy(updated = now))
+      }
+    }
+
+
+  final def inTransactionT[A](action: DataAccess => ReadWriteAction[Option[A]]): MigrateAction[A] =
+    inTransaction(action).mapF(optT => OptionT(optT.value.map(_.flatten)))
+
+
+  final def inTransaction[A](action: DataAccess => ReadWriteAction[A]): MigrateAction[A] =
+    for {
+      dataSource <- MigrateAction.asks(_.dataSource)
+      result <- MigrateAction.liftIO(dataSource.inTransaction(action).io)
+    } yield result
+
+
+  final def getWorkspace(workspaceId: UUID): MigrateAction[Workspace] =
+    inTransaction {
+      _.workspaceQuery.findByIdOrFail(workspaceId.toString)
+    }
+
+
+  final def nowTimestamp: MigrateAction[Timestamp] =
+    MigrateAction.liftIO(IO {
+      Timestamp.valueOf(LocalDateTime.ofInstant(Instant.now, ZoneOffset.UTC))
+    })
+
+
+  final def randomSuffix(str: String): IO[String] = IO {
+    str ++ UUID.randomUUID.toString.replace("-", "")
+  }
+
+
+  final def noGoogleProjectError[A](migration: WorkspaceMigration): Throwable =
+    WorkspaceMigrationException(
+      message = "Workspace migration failed: Google Project not found.",
+      data = Map(
+        ("migrationId" -> migration.id),
+        ("workspaceId" -> migration.workspaceId),
+        ("googleProjectId" -> migration.newGoogleProjectId)
+      )
+    )
+
+
+  final def noWorkspaceBucketError[A](migration: WorkspaceMigration, bucket: GcsBucketName): Throwable =
+    WorkspaceMigrationException(
+      message = "Workspace migration failed: Workspace cloud bucket not found.",
+      data = Map(
+        ("migrationId" -> migration.id),
+        ("workspaceId" -> migration.workspaceId),
+        ("workspaceBucket" -> bucket)
+      )
+    )
+
+
+  final def noTmpBucketError[A](migration: WorkspaceMigration): Throwable =
+    WorkspaceMigrationException(
+      message = "Workspace migration failed: Temporary cloud storage bucket not found.",
+      data = Map(
+        ("migrationId" -> migration.id),
+        ("workspaceId" -> migration.workspaceId),
+        ("tmpBucket" -> migration.tmpBucketName)
+      )
+    )
+
+
+  sealed trait Message
+  final case class Schedule(workspace: Workspace) extends Message
+  case object RunMigration extends Message
+  case object RefreshTransferJobs extends Message
+
+  // ehigham: I've included this actor implementation as it's helped inform how to refresh
+  // the status of storage transfer jobs. While it's as yet untested, this can be plumbed-in
+  // and integration tested as part of CA-1132.
+  // todo: finalise design and test in CA-1132
+  def apply(pollingInterval: FiniteDuration,
+            dataSource: SlickDataSource,
+            googleProjectToBill: GoogleProject,
+            workspaceService: WorkspaceService,
+            storageService: GoogleStorageService[IO],
+            storageTransferService: GoogleStorageTransferService[IO])
+  : Behavior[Message] =
+    Behaviors.setup { context =>
```

Comment:
Q: Behaviors?

Reply:
I can only point you to the akka doc. I have no clue.